MediaRecorder.html

```diff
@@ -222,61 +223,83 @@
-                  [[!FILE-API]]), and queue a task, using the DOM manipulation
-                  task source, to fire an event named <code>start</code>.</li>
+                  stream's tracks, let <var>blob</var> be the Blob
+                  ([[!FILE-API]]) of collected data, start gathering the data
+                  into <var>blob</var> and queue a task, using the DOM
+                  manipulation task source, to <a
+                  href="https://dom.spec.whatwg.org/#concept-event-fire">fire
+                  </a> an event named <code>start</code>.</li>
 
                   <li>If at any point the stream's isolation properties change
                   so that MediaRecorder is no longer allowed access to it, the
                   UA MUST immediately stop gathering data, discard any data that
                   it has gathered, and queue a task, using the DOM manipulation
-                  task source, that <a>fires an error event</a> with the name
-                  <code>SecurityError</code>, immediately followed by a <code>
-                  stop</code> event. </li>
+                  task source, that runs the following steps:
+                  <ol>
+                    <li>Set <a>state</a> to
+                    <a href="#dom-recordingstate-inactive">inactive</a>.</li>
+                    <li><a href="#dfn-firing-an-error-event-named-e">Fire an
+                    error event</a> with the name <code>SecurityError</code>
+                    </li>
+                    <li><a href="https://dom.spec.whatwg.org/#concept-event-fire">
+                    Fire</a> an event named <code>stop</code>.</li>
+                  </ol>
+                  </li>
 
                   <li>If the UA at any point is unable to continue gathering
                   data for reasons other than isolation properties, it MUST stop
                   gathering data, let <var>blob</var> be the Blob of collected
                   data, and queue a task, using the DOM manipulation task
-                  source, that <a>fires an error event</a> with the name
-                  <code>UnknownError</code>, followed by a
-                  <code>dataavailable</code> event containing <var>blob</var>,
-                  followed by a <code>stop</code> event. </li>
+                  source, that runs the following steps:
+                  <ol>
+                    <li>Set <a>state</a> to
+                    <a href="#dom-recordingstate-inactive">inactive</a>.</li>
+                    <li><a href="#dfn-firing-an-error-event-named-e">Fire an
+                    error event</a> with the name <code>UnknownError</code>
+                    </li>
+                    <li><a href="#dfn-firing-a-dataavailable-event">Fire a
+                    dataavailable event</a> with <var>blob</var>.</li>
+                    <li><a href="https://dom.spec.whatwg.org/#concept-event-fire">
+                    Fire</a> an event named <code>stop</code>.</li>
```

Comment:
Here and elsewhere I'd suggest making all of "fire an event" the linked text. It doesn't really matter, but for consistency and if you ever convert this to Bikeshed that's what you'll use to get the linked right automatically.